### PR TITLE
Check only local groups when adding new supplementary groups to a user

### PR DESCRIPTION
**Pull request by ikerexxe**
usermod and useradd: check only local groups with -G option

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1727236

### CODE REVIEW

File: src/usermod.c [hunk 1]
```diff
@@ -328,12 +336,52 @@ static int get_groups (char *list)
 	 * Any errors in finding group names are fatal
 	 */
 	if (0 != errors) {
+		close_group_files ();
 		return -1;
 	}
 
+	close_group_files ();
```
Comment (t8m) [inline src/usermod.c]: Would it make sense to avoid the duplication here and move the close before the if()?

File: src/useradd.c [hunk 2]
```diff
@@ -805,12 +814,54 @@ static int get_groups (char *list)
 	 * Any errors in finding group names are fatal
 	 */
 	if (0 != errors) {
+		close_group_files ();
```
Comment (t8m) [inline src/useradd.c]: And here as well I'd avoid the duplication and move the close/unlock before the if().

File: src/usermod.c (hunk 1 above)
Comment (ikerexxe) [reply src/usermod.c]: As you say it makes more sense to close the files before the if(), in order to minimize future errors.

File: src/useradd.c (hunk 2 above)
Comment (ikerexxe) [reply src/useradd.c]: As you say it makes more sense to close the files before the if(), in order to minimize future errors. I have just pushed the changes.

**Comment by ikerexxe** [comment]
@hallyn could you please review this change?

**Comment by hallyn** [comment]
Thanks, but it''s hard to review as is.  The only explanation of why is a pointer to https://bugzilla.redhat.com/show_bug.cgi?id=1727236 , but I'm not authorized to see that.

Please add a detailed rationale in the commit message.

**Comment by t8m** [comment]
@hallyn I've opened up the RHEL bug report so you should be able to read it now.

**Comment by hallyn** [comment]
> @hallyn I've opened up the RHEL bug report so you should be able to read it now.

Thanks.

**Comment by hallyn** [comment]
It seems like these functions that are repeated in useradd and usermod could be done in lib/ or libmisc/ instead, right?

**Comment by hallyn** [comment]
> It seems like these functions that are repeated in useradd and usermod could be done in lib/ or libmisc/ instead, right?

never mind, i see, i should probably restructure that set of files myself.

**Comment by hallyn** [comment]
Thanks - IIUC this also fixes the FIXME in useradd.c following the old prefix_getgr_nam_gid call, right?

**Comment by ikerexxe** [comment]
Yes, it's also fixed as this change only allows local groups to be added as supplementary groups.

**Pull request merged**